src/uu/tr/src/tr.rs

```diff
@@ -170,56 +181,60 @@ fn translate_input<T: SymbolTranslator>(
     }
 }
 
-fn usage(opts: &Options) {
-    println!("{} {}", NAME, VERSION);
-    println!();
-    println!("Usage:");
-    println!("  {} [OPTIONS] SET1 [SET2]", NAME);
-    println!();
-    println!("{}", opts.usage("Translate or delete characters."));
+fn get_usage() -> String {
+    format!("{} [OPTION]... SET1 [SET2]", executable!())
 }
 
 pub fn uumain(args: impl uucore::Args) -> i32 {
-    let args = args.collect_str();
-
-    let mut opts = Options::new();
-
-    opts.optflag("c", "complement", "use the complement of SET1");
-    opts.optflag("C", "", "same as -c");
-    opts.optflag("d", "delete", "delete characters in SET1");
-    opts.optflag("h", "help", "display this help and exit");
-    opts.optflag("s", "squeeze", "replace each sequence of a repeated character that is listed in the last specified SET, with a single occurrence of that character");
-    opts.optflag(
-        "t",
-        "truncate-set1",
-        "first truncate SET1 to length of SET2",
-    );
-    opts.optflag("V", "version", "output version information and exit");
-
-    let matches = match opts.parse(&args[1..]) {
-        Ok(m) => m,
-        Err(err) => {
-            show_error!("{}", err);
-            return 1;
-        }
+    let usage = get_usage();
+
+    let matches = App::new(executable!())
+        .version(VERSION)
+        .about(ABOUT)
+        .usage(&usage[..])
+        .after_help(LONG_HELP)
+        .arg(
+            Arg::with_name(options::COMPLEMENT)
+                .short("C")
+                .short("c")
+                .long(options::COMPLEMENT)
+                .help("use the complement of SET1"),
+        )
+        .arg(
+            Arg::with_name(options::DELETE)
+                .short("d")
+                .long(options::DELETE)
+                .help("delete characters in SET1, do not translate"),
+        )
+        .arg(
+            Arg::with_name(options::SQUEEZE)
+                .long(options::SQUEEZE)
+                .short("s")
+                .help(
+                    "replace each sequence  of  a  repeated  character  that  is
+            listed  in the last specified SET, with a single occurrence
+            of that character",
+                ),
+        )
+        .arg(
+            Arg::with_name(options::TRUNCATE)
+                .long(options::TRUNCATE)
+                .short("t")
+                .help("first truncate SET1 to length of SET2"),
+        )
+        .arg(Arg::with_name(options::SETS).multiple(true))
+        .get_matches_from(args);
+
+    let dflag = matches.is_present(options::DELETE);
+    let cflag = matches.is_present(options::COMPLEMENT);
+    let sflag = matches.is_present(options::SQUEEZE);
+    let tflag = matches.is_present(options::TRUNCATE);
+
+    let sets: Vec<String> = match matches.values_of(options::SETS) {
+        Some(v) => v.map(|v| v.to_string()).collect(),
+        None => vec!["".to_string()],
```

Comment:
Using a empty string inside this will cause to program hang forever since it should fail on this line

```rust
  if sets.is_empty() {
        show_error!(
            "missing operand\nTry `{} --help` for more information.",
            NAME
        );
        return 1;
    }
```


changing this line to `None => vec![],` should do good.

Reply:
Thank you for letting me know, i have made the change